firedrake/slate/slate.py

```diff
@@ -635,14 +738,19 @@ def __init__(self, A):
         assert A.shape[0] == A.shape[1], (
             "The inverse can only be computed on square tensors."
         )
+
+        if A.shape > (4, 4) and not isinstance(A, Factorization):
+            A = Factorization(A, decomposition="PartialPivLU")
+
```

Comment:
This works because eigen turns `Factorisation(A).inverse()*foo` into "apply factorisation of `A` to `foo`, right?

Reply:
Correct. In Eigen:
```cpp
A.inverse() * b;
```
is mathematically equivalent to
```cpp
Eigen::PartialPivLU< ... > dec(A);
dec.inverse() * b;  // same as dec.solve(b)
```
In the latter case, the inverse of `A` is not explicitly computed; this just applies the already declared factorization.